### PR TITLE
Promotion of binary-ops to variadic-ops.

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: lib-clay/deques/deques.clay [hunk 1]
```diff
@@ -79,7 +79,7 @@ record DequeCoordinate[T] (
 }
 
 [T] overload subtract(i: DequeCoordinate[T], j: DequeCoordinate[T]) : PtrInt {
-    var mapDistance = PtrInt(i.node - j.node - 1);
+    var mapDistance = PtrInt((i.node - j.node) - 1);
```
Comment (jckarter) [inline lib-clay/deques/deques.clay]: Can this change be backed out now?

Comment (ghost) [inline lib-clay/deques/deques.clay]: Hmm, no. I think the problem here is all variadic args are expected to 
be of the same type. Not sure of the best way to fix this . .

Comment (jckarter) [inline lib-clay/deques/deques.clay]: The type of the args shouldn't matter, since the variadic overload just forwards to the binary overload.

Comment (ghost) [inline lib-clay/deques/deques.clay]: Ha, daft copy/paste error - left the return type in the arithmetic 
overloads. Think this will solve it.

**Pull request closed without merging**